### PR TITLE
remove getrandom::register_custom_getrandom

### PR DESCRIPTION
**Pull request by lastmjs**
My main concern with merging this is forcing Azle to switch to Wasi possibly before we're ready. So either this stays on its own branch or we need to put some conditional logic in.

@bdemann @dansteren what do you think?

### CODE REVIEW

**Comment by dansteren** [comment]
@lastmjs I agree that this shouldn't be merged into main as is at this time, because wasi support isn't even fully out (as far as I understand).

I think I generally don't like long lived branches and having logic that understands the target and only removes this on wasi seems nice to me, so I would get behind that plan. That being said, the changes here are relatively small so just closing this PR for now and reopening it later also doesn't seem like a bad plan. So, either option seems fine. But yeah, let's hold off on merging it as is.

**Comment by lastmjs** [comment]
I should use conditional compilation to keep stuff for wasm32-unknown-unknown until we have true full wasi support

**Review by dansteren — DISMISSED** [review]
Very nice!

File: src/act/random.rs [hunk 1]
```diff
@@ -36,5 +40,21 @@ pub fn generate_randomness_implementation() -> TokenStream {
                 });
             });
         }
+
+        #[cfg(all(target_arch = "wasm32", target_os = "wasi"))]
+        fn rng_seed() {
+            ic_cdk::spawn(async move {
+                let result: ic_cdk::api::call::CallResult<(Vec<u8>,)> = ic_cdk::api::call::call(
+                    candid::Principal::from_text("aaaaa-aa").unwrap(),
```
Comment (bdemann) [inline src/act/random.rs:48]: 😱

**Review by bdemann — CHANGES_REQUESTED** [review]

File: src/act/random.rs (hunk 1 above)
Comment (lastmjs) [reply src/act/random.rs:48]: We don't have runtime error infrastructure for the cdk_framework yet, so I'm punting this to whoever does that

Comment (lastmjs) [reply src/act/random.rs]: I've removed two unwraps by using the management_canister API directly, the rest were already there and we haven't done a pass to figure out runtime errors in the cdk_framework yet

**Review by bdemann — APPROVED** [review]

**Pull request merged**